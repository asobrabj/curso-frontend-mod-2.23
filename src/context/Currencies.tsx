import { createContext, useContext, useState } from 'react';
import instance from '../service/axiosaInstace';

export interface ICurrencyContext {
  to: string;
  setTo: React.Dispatch<React.SetStateAction<string>>;
  from: string;
  setFrom: React.Dispatch<React.SetStateAction<string>>;
  amount: string;
  setAmount: React.Dispatch<React.SetStateAction<string>>;
  fecthConversion: (
    to: string,
    from: string,
    amount: string,
  ) => Promise<string>;
  error: boolean;
}

interface ICurrencyProvider {
  children: React.ReactNode;
}

const CurrencyContext = createContext<ICurrencyContext>({} as ICurrencyContext);

export const CurrencyProvider: React.FC<ICurrencyProvider> = ({ children }) => {
  const [error, setError] = useState<boolean>(false);
  const [to, setTo] = useState<string>('');
  const [from, setFrom] = useState<string>('');
  const [amount, setAmount] = useState<string>('');

  const fecthConversion = async (
    to: string,
    from: string,
    amount: string,
  ): Promise<string> => {
    try {
      const response = await instance.get(`${to}/${from}/${amount}`);
      return response.data.conversion_result;
    } catch (err) {
      setError(true);
      return 'Erro na convers√£o';
    }
  };

  return (
    <CurrencyContext.Provider
      value={{
        fecthConversion,
        error,
        amount,
        from,
        to,
        setTo,
        setAmount,
        setFrom,
      }}
    >
      {children}
    </CurrencyContext.Provider>
  );
};

export const useCurrency = () => {
  return useContext(CurrencyContext);
};
