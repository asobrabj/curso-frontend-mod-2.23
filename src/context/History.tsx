/* eslint-disable react-refresh/only-export-components */
import { createContext, useContext, useEffect, useState } from 'react';

export interface IDataCurrency {
  to: string;
  from: string;
  amount: string;
  result: string;
  date: string;
}

export interface IHistoryContext {
  setHistory: (data: IDataCurrency) => void;
  history?: Array<IDataCurrency>;
}

interface IHistoryProvider {
  children: React.ReactNode;
}

const HistoryContext = createContext<IHistoryContext>({} as IHistoryContext);

export const HistoryProvider: React.FC<IHistoryProvider> = ({ children }) => {
  const [history, setHistoryState] = useState<Array<IDataCurrency>>();

  useEffect(() => {
    const currentData = window.localStorage.getItem('history');
    const newData = currentData ? JSON.parse(currentData) : [];
    setHistoryState(newData);
  }, []);

  const setHistory = (data: IDataCurrency) => {
    const currentData = window.localStorage.getItem('history');
    const newData = currentData ? JSON.parse(currentData) : [];
    newData.push(data);
    setHistoryState(newData);
    window.localStorage.setItem('history', JSON.stringify(newData));
    return;
  };

  return (
    <HistoryContext.Provider value={{ setHistory, history }}>
      {children}
    </HistoryContext.Provider>
  );
};

export const useHistory = () => {
  return useContext(HistoryContext);
};
