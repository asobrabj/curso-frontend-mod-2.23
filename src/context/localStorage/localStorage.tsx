import { createContext, useContext, useEffect, useState } from 'react'
import { IFormData } from '../../components/FormAddAnnotation'
import IList from '../../types/lists'

export interface ILocalStorageContext {
  anotations: Array<IFormData> | null
  setAnotations: (data: IFormData | Array<IFormData>) => void
  getFilters: (value: string) => void
  getTags: (key: 'category' | 'priority', value: string) => void
  getStorage: (key: string) => Array<IFormData | IList> | string | null
  setStorage: (key: string, data: Array<IFormData | IList> | string) => void
}

interface ILocalStorageProvider {
  children: React.ReactNode
}

const LocalStorageContext = createContext<ILocalStorageContext>(
  {} as ILocalStorageContext
)

export function LocalStorageProvider({ children }: ILocalStorageProvider) {
  const [anotations, setAnotations] = useState<Array<IFormData> | null>([])

  useEffect(() => {
    const anotations = getStorage('anotations')
    setAnotations(anotations)
  }, [])

  const getTags = (key: 'category' | 'priority', value: string) => {
    const anotations = getStorage('anotations')
    const data = anotations?.filter(
      (item: IFormData) =>
        item[key]?.toLowerCase().includes(value.toLowerCase())
    )

    if (data) setAnotations(data)
    return
  }

  const getFilters = (value: string) => {
    const anotations = getStorage('anotations')
    const name = anotations?.filter((item: IFormData) =>
      item.name.toLowerCase().includes(value.toLowerCase())
    )

    if (name) setAnotations(name)
    return
  }

  const setAnotationsFn = (data: IFormData | Array<IFormData>) => {
    const storage = getStorage('anotations')
    let newStorage: Array<IFormData> = []
    if (storage) {
      newStorage = [...storage, data]
    }

    if (data instanceof Array) {
      newStorage = data
    }

    setAnotations(newStorage)
    localStorage.setItem('anotations', JSON.stringify(newStorage))
  }

  const getStorage = (key: string) => {
    const storedData = localStorage.getItem(key)
    if (storedData) {
      return JSON.parse(storedData)
    }
    return []
  }

  const saveToStorage = (
    key: string,
    data: Array<IFormData | IList> | string
  ) => {
    localStorage.setItem(key, JSON.stringify(data))
    return
  }
  return (
    <LocalStorageContext.Provider
      value={{
        getTags,
        getFilters,
        anotations,
        setAnotations: setAnotationsFn,
        getStorage,
        setStorage: saveToStorage,
      }}
    >
      {children}
    </LocalStorageContext.Provider>
  )
}

export function useStorage() {
  return useContext(LocalStorageContext)
}
