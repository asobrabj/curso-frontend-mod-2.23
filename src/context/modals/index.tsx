import { createContext, useContext, useState } from 'react'
import IAnotation from '../../types/anotation'

export interface IModalContext {
  isVisible: boolean
  setIsVisible: React.Dispatch<React.SetStateAction<boolean>>
  isForm: boolean
  setIsForm: React.Dispatch<React.SetStateAction<boolean>>
  dataCard: IAnotation
  setDataCard: React.Dispatch<React.SetStateAction<IAnotation>>
  idCard: number | null
  setIdCard: React.Dispatch<React.SetStateAction<number | null>>
}

interface IModalProvider {
  children: React.ReactNode
}

const ModalContext = createContext<IModalContext>({} as IModalContext)

export function ModalProvider({ children }: IModalProvider) {
  const [isVisible, setIsVisible] = useState<boolean>(false)
  const [isForm, setIsForm] = useState<boolean>(false)
  const [idCard, setIdCard] = useState<number | null>(null)
  const [dataCard, setDataCard] = useState<IAnotation>({
    id: null,
    concluded: false,
    date: '',
    description: '',
    name: '',
    category: '',
    priority: '',
    time: '',
  })

  return (
    <ModalContext.Provider
      value={{
        idCard,
        setIdCard,
        isVisible,
        setIsVisible,
        isForm,
        setIsForm,
        dataCard,
        setDataCard,
      }}
    >
      {children}
    </ModalContext.Provider>
  )
}

export function useModal() {
  return useContext(ModalContext)
}
