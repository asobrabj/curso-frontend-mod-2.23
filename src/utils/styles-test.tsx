import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import theme from '../styles/darkTheme'

export const renderTheme = (Children: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}> {Children} </ThemeProvider>)
}
