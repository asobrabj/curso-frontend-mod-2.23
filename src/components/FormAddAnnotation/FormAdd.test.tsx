import {
  act,
  fireEvent,
  renderHook,
  screen,
  waitFor,
} from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import FormAddAnotation, { IFormData } from '.'
import { useStorage } from '../../context/localStorage/localStorage'
import { useModal } from '../../context/modals'
import { renderTheme } from '../../utils/styles-test'

jest.mock('../../context/modals', () => ({
  useModal: jest.fn(() => ({
    setIsVisible: jest.fn(),
    idCard: null,
    setIdCard: jest.fn(),
    isVisible: true,
  })),
}))

jest.mock('../../context/localStorage/localStorage', () => ({
  useStorage: jest.fn(() => ({
    anotations: [],
    setAnotations: jest.fn(),
    getStorage: jest.fn(),
  })),
}))

const mock: IFormData[] = [
  {
    id: 0,
    name: 'teste',
    date: '2023-12-11',
    time: '00:00',
    description: ' um texto qualquer',
    concluded: false,
    category: 'estudo',
    priority: 'media',
  },
  {
    id: 1,
    name: 'testando',
    date: '2023-12-11',
    time: '00:00',
    description: ' um texto qualquer',
    concluded: false,
    category: 'finanças',
    priority: 'alta',
  },
]

describe('FormAddAnotation Component', () => {
  test('Renderização Inicial', () => {
    const { getByText } = renderTheme(<FormAddAnotation />)

    expect(getByText('formulario')).toBeInTheDocument()
    expect(getByText(/Nome/i)).toBeInTheDocument()
    expect(getByText('Salvar')).toBeInTheDocument()
    expect(getByText('Cancelar')).toBeInTheDocument()
  })

  test('Envio do Formulário', async () => {
    renderTheme(<FormAddAnotation />)

    const { result: resultModal } = renderHook(() => useModal())
    const { result: resultStorage } = renderHook(() => useStorage())

    const nameInput = screen.getByRole('textbox', {
      name: /Nome/i,
    }) as HTMLInputElement
    const dateInput = screen.getByTitle('title-date') as HTMLInputElement
    const timeInput = screen.getByTitle('title-time') as HTMLInputElement
    const selectCategory = screen.getByTestId(
      'select-category'
    ) as HTMLSelectElement
    const selectPriority = screen.getByTestId(
      'select-priority'
    ) as HTMLSelectElement
    expect(selectCategory).toBeInTheDocument()
    expect(selectPriority).toBeInTheDocument()
    expect(dateInput).toBeInTheDocument()
    expect(timeInput).toBeInTheDocument()
    expect(screen.getByText('Salvar')).toBeInTheDocument()
    const saveButton = screen.getByText('Salvar')

    act(() => {
      fireEvent.change(nameInput, { target: { value: 'Teste Nome' } })
      fireEvent.change(timeInput, { target: { value: '00:00' } })
      fireEvent.change(dateInput, { target: { value: '2023-12-25' } })
      fireEvent.click(saveButton)
    })

    expect(nameInput.value).toBe('Teste Nome')
    expect(timeInput.value).toBe('00:00')
    expect(dateInput.value).toBe('2023-12-25')

    waitFor(() => {
      expect(resultStorage.current.setAnotations).toHaveBeenCalledTimes(1)
      expect(resultModal.current.setIsVisible).toHaveBeenCalledTimes(1)
      expect(resultModal.current.setIsForm).toHaveBeenCalledTimes(1)
      expect(resultModal.current.isVisible).toBe(false)
    })
  })

  it('botao de cancelamento', () => {
    renderTheme(<FormAddAnotation />)
    const btnCancel = screen.getByRole('button', { name: 'Cancelar' })
    expect(btnCancel).toBeInTheDocument()

    act(() => {
      fireEvent.click(btnCancel)
    })
    waitFor(() => {
      const mockUseModal = useModal()
      expect(mockUseModal.setIsVisible).toHaveBeenCalled()
      expect(mockUseModal.setIdCard).toHaveBeenCalled()
      expect(mockUseModal.isVisible).toBe(true)
    })
  })

  it('icone para fechar formulario', () => {
    renderTheme(<FormAddAnotation />)
    const icon = screen.getByTitle('close form')
    act(() => {
      fireEvent.click(icon)
    })

    waitFor(() => {
      const mockUseModal = useModal()
      expect(mockUseModal.setIsVisible).toHaveBeenCalled()
      expect(mockUseModal.isVisible).toBe(true)
    })
  })

  test('Envio do Formulário com data errada', async () => {
    renderTheme(<FormAddAnotation />)

    const { result: resultModal } = renderHook(() => useModal())

    const nameInput = screen.getByRole('textbox', {
      name: /Nome/i,
    }) as HTMLInputElement
    const dateInput = screen.getByText(/Data/i) as HTMLInputElement
    const timeInput = screen.getByText(/Hora/i) as HTMLInputElement
    const selectCategory = screen.getByTestId(
      'select-category'
    ) as HTMLSelectElement
    const selectPriority = screen.getByTestId(
      'select-priority'
    ) as HTMLSelectElement
    expect(selectCategory).toBeInTheDocument()
    expect(selectPriority).toBeInTheDocument()
    expect(screen.getByText('Salvar')).toBeInTheDocument()
    const saveButton = screen.getByText('Salvar')
    const optionCategory = screen.getAllByTestId('options')[0]
    const optionPriority = screen.getAllByTestId('options')[6]

    act(() => {
      userEvent.type(nameInput, 'Teste Nome')
      userEvent.type(dateInput, '2022-11-31')
      userEvent.type(timeInput, '15:30')
      fireEvent.click(optionCategory)
      fireEvent.click(optionPriority)
      fireEvent.click(saveButton)
    })

    waitFor(() => {
      const error = screen.getByText(
        'Não é possivel selecionar uma data passada'
      )
      expect(error).toBeInTheDocument()
      expect(resultModal.current.isVisible).toBe(true)
      expect(optionPriority.textContent).toBe('media')
      expect(nameInput.value).toBe('Teste Nome')
      expect(dateInput.value).toBe('2023-12-31')
      expect(timeInput.value).toBe('15:30')
      expect(optionCategory.textContent).toBe('finanças')
    })
  })

  it('adicionando nova anotação', () => {
    localStorage.clear()
    renderTheme(<FormAddAnotation />)
    const { result } = renderHook(() => useStorage())
    const nameInput = screen.getByRole('textbox', {
      name: /Nome/i,
    }) as HTMLInputElement
    const dateInput = screen.getByText(/Data/i) as HTMLInputElement
    const timeInput = screen.getByText(/Hora/i) as HTMLInputElement
    const selectCategory = screen.getByTestId(
      'select-category'
    ) as HTMLSelectElement
    const selectPriority = screen.getByTestId(
      'select-priority'
    ) as HTMLSelectElement
    expect(selectCategory).toBeInTheDocument()
    expect(selectPriority).toBeInTheDocument()
    expect(screen.getByText('Salvar')).toBeInTheDocument()
    const saveButton = screen.getByText('Salvar')
    const optionCategory = screen.getAllByTestId('options')[0]
    const optionPriority = screen.getAllByTestId('options')[6]

    act(() => {
      userEvent.type(nameInput, 'Teste Nome')
      userEvent.type(dateInput, '2023-12-31')
      userEvent.type(timeInput, '15:30')
      fireEvent.click(optionCategory)
      fireEvent.click(optionPriority)
      fireEvent.click(saveButton)
    })

    waitFor(() => {
      expect(result.current.setAnotations).toHaveBeenCalled()
    })
  })
  it('inputNome', async () => {
    act(() => {
      renderTheme(<FormAddAnotation />)
    })
    const nameInput = screen.getByRole('textbox', {
      name: /Nome/i,
    }) as HTMLInputElement

    expect(nameInput).toBeInTheDocument()
    expect(nameInput.value).toBe('')

    act(() => {
      fireEvent.change(nameInput, { target: { value: 'ab' } })
    })
    waitFor(() => {
      const input = screen.getByRole('textbox', {
        name: /Nome/i,
      }) as HTMLInputElement

      expect(input.value).toBe('ab')
    })
  })

  it('inputData', async () => {
    act(() => {
      renderTheme(<FormAddAnotation />)
    })
    const dataInput = screen.getByTitle('title-date') as HTMLInputElement

    expect(dataInput).toBeInTheDocument()

    act(() => {
      fireEvent.change(dataInput, { target: { value: '2023-12-31' } })
    })
    expect(dataInput.value).toBe('2023-12-31')
  })

  it('inputTime', async () => {
    act(() => {
      renderTheme(<FormAddAnotation />)
    })
    const timeInput = screen.getByLabelText(/Hora/i)

    expect(timeInput).toBeInTheDocument()

    act(() => {
      fireEvent.change(timeInput, { target: { value: '00:00' } })
    })
    expect(timeInput.value).toBe('00:00')
  })

  it('submeter formulario sem dados', () => {
    const { container } = renderTheme(<FormAddAnotation />)
    const { result } = renderHook(() => useModal())
    const btn = screen.getByText('Salvar')
    expect(container).toBeInTheDocument()
    act(() => {
      fireEvent.click(btn)
    })
    expect(container).toBeInTheDocument()

    expect(result.current.setIsVisible).not.toHaveBeenCalled()
    expect(result.current.setIdCard).not.toHaveBeenCalled()
    expect(result.current.isVisible).toBe(true)
  })

  it('textArea', () => {
    renderTheme(<FormAddAnotation />)
    const textarea = screen.getByTitle('message') as HTMLTextAreaElement
    act(() => {
      fireEvent.change(textarea, {
        target: { value: 'texto' },
      })
    })

    expect(textarea.value).toBe('texto')
  })

  it('submeter formulario data passada', async () => {
    renderTheme(<FormAddAnotation />)
    const btn = screen.getByText('Salvar')
    const nameInput = screen.getByRole('textbox', {
      name: /Nome/i,
    }) as HTMLInputElement
    const dateInput = screen.getByTitle('title-date') as HTMLInputElement
    const timeInput = screen.getByTitle('title-time') as HTMLInputElement

    act(() => {
      fireEvent.change(nameInput, { target: { value: 'Teste Nome' } })
      fireEvent.change(timeInput, { target: { value: '00:00' } })
      fireEvent.change(dateInput, { target: { value: '2023-10-25' } })
      fireEvent.click(btn)
    })

    const error = await screen.findByText(
      'Não é possivel selecionar uma data passada'
    )

    expect(error).toBeInTheDocument()
  })

  it('textArea', () => {
    renderTheme(<FormAddAnotation />)
    const textarea = screen.getByTitle('message') as HTMLTextAreaElement
    act(() => {
      fireEvent.change(textarea, {
        target: { value: 'texto' },
      })
    })

    expect(textarea.value).toBe('texto')
  })

  it('snapshot', () => {
    const { container } = renderTheme(<FormAddAnotation />)
    expect(container).toMatchSnapshot()
  })
})
