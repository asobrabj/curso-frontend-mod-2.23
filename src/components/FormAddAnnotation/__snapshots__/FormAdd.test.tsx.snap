// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormAddAnotation Component snapshot 1`] = `
.c5 {
  position: absolute;
  top: -10px;
  left: 10px;
}

.c5 span {
  color: red;
  margin-right: 3px;
}

.c7 {
  padding: 8px;
  border: 1px solid #00abff;
  margin: 1px;
  width: 100%;
  background-color: '';
}

.c7:focus {
  outline: none;
  border: 2px solid #00abff;
  margin: 0;
}

.c4 {
  margin-bottom: 25px;
  position: relative;
  max-width: 350px;
  width: 100%;
}

.c4.primary .c6 {
  border-radius: 8px;
}

.c4.secondary .c6 {
  border-radius: 8px 0;
}

.c4.error {
  color: #ff0000;
}

.c4.error .c6 {
  border: 1px solid #ff0000;
}

.c4.error .c6:focus {
  border: 2px solid #ff0000;
}

.c9 {
  margin-bottom: 25px;
  position: relative;
  max-width: 150px;
  width: 100%;
}

.c9.primary .c6 {
  border-radius: 8px;
}

.c9.secondary .c6 {
  border-radius: 8px 0;
}

.c9.error {
  color: #ff0000;
}

.c9.error .c6 {
  border: 1px solid #ff0000;
}

.c9.error .c6:focus {
  border: 2px solid #ff0000;
}

.c11 {
  position: absolute;
  top: -12px;
  left: 12px;
  background-color: '';
}

.c11 span {
  font-size: 14px;
  color: red;
  margin-right: 3px;
}

.c10 {
  text-transform: capitalize;
  margin-bottom: 25px;
  width: 150px;
  border: 1px solid #00abff;
  position: relative;
}

.c10.primary {
  width: 150px;
  padding: 8px;
  border-radius: 8px;
}

.c10.secondary {
  width: 150px;
  padding: 8px;
  border-radius: 8px 0;
}

.c10.error {
  border: 1px solid #ff0000;
  color: #ff0000;
}

.c10.error:focus {
  border: 2px solid #ff0000;
}

.c10 select {
  display: none;
}

.c10.custom-select {
  position: relative;
  display: inline-block;
  width: 100%;
}

.c10 .custom-select-trigger {
  width: 100%;
  cursor: pointer;
  height: 20px;
}

.c10 .custom-options {
  background-color: '';
  display: none;
  position: absolute;
  top: 100%;
  left: 0px;
  right: 0;
  background-color: #fff;
  border: 1px solid #00abff;
  max-width: 150px;
  width: 100%;
  max-height: 114px;
  overflow-y: auto;
  z-index: 1;
}

.c10 .custom-options.primary {
  border-radius: 8px;
}

.c10 .custom-options.secondary {
  border-radius: 8px 0;
}

.c10 .custom-option {
  padding: 10px;
  cursor: pointer;
}

.c10 .custom-option:hover {
  background-color: '';
  color: #00abff;
}

.c15 {
  padding: 8px;
  border: 1px solid #00abff;
  margin: 1px;
  resize: none;
  width: 100%;
  height: 100%;
}

.c15:focus {
  outline: none;
  border: 2px solid #00abff;
  margin: 0;
}

.c13 {
  position: absolute;
  top: -10px;
  left: 12px;
  background-color: '';
}

.c13 span {
  font-size: 14px;
  color: red;
  margin-right: 3px;
}

.c12 {
  margin-bottom: 25px;
  max-width: 350px;
  height: 120px;
  position: relative;
}

.c12.primary .c14 {
  border-radius: 8px;
}

.c12.secondary .c14 {
  border-radius: 8px 0;
}

.c12.error {
  color: #ff0000;
}

.c12.error .c14 {
  border: 1px solid #ff0000;
}

.c12.error .c14:focus {
  border: 2px solid #ff0000;
}

.c16 {
  padding: 8px;
  border-radius: 8px;
  margin-right: 16px;
  font-weight: bold;
  cursor: pointer;
}

.c0 {
  position: relative;
  padding: 16px;
  height: 500px;
  width: 350px;
  border-radius: 16px;
  border: 1px solid #999;
}

.c1 {
  position: absolute;
  right: 8px;
  top: 5px;
  font-size: 24px;
  cursor: pointer;
  padding: 8px;
  color: #7a1616;
}

.c2 {
  display: flex;
  flex-direction: column;
}

.c3 {
  font-style: italic;
  text-align: center;
  margin-bottom: 24px;
}

.c8 {
  display: flex;
  justify-content: space-between;
}

.c17 {
  background-color: #2c6600;
}

.c18 {
  background-color: #7a1616;
}

<div>
   
  <div
    class="c0"
  >
    <i
      class="c1"
      title="close form"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"
        />
      </svg>
    </i>
    <form
      class="c2"
    >
      <h2
        class="c3"
      >
        formulario
      </h2>
      <div
        class="c4 primary undefined"
      >
        <label
          class="c5"
          for="name-id"
        >
          <span>
            *
          </span>
          Nome
        </label>
        <input
          class="c6 c7"
          id="name-id"
          name="name"
          required=""
          title="title-name"
          type="text"
          value=""
        />
      </div>
      <div
        class="c8"
      >
        <div
          class="c9 primary undefined"
        >
          <label
            class="c5"
            for="date-id"
          >
            <span>
              *
            </span>
            Data
          </label>
          <input
            class="c6 c7"
            id="date-id"
            name="date"
            required=""
            title="title-date"
            type="date"
            value=""
          />
        </div>
        <div
          class="c9 primary undefined"
        >
          <label
            class="c5"
            for="time-id"
          >
            <span>
              *
            </span>
            Hora
          </label>
          <input
            class="c6 c7"
            id="time-id"
            name="time"
            required=""
            title="title-time"
            type="time"
            value=""
          />
        </div>
      </div>
      <div
        class="c8"
      >
        <div
          class="c10 primary undefined"
        >
          <div
            class="custom-select"
          >
            <label
              class="c11"
              for="category"
            >
               
              Categoria
            </label>
            <div
              class="custom-select-trigger primary undefined"
              data-testid="select-category"
              id="category"
            />
            <div
              class="custom-options primary undefined"
            >
              <div
                class="custom-option"
                data-testid="options"
                data-value="finanças"
              >
                finanças
              </div>
              <div
                class="custom-option"
                data-testid="options"
                data-value="comida"
              >
                comida
              </div>
              <div
                class="custom-option"
                data-testid="options"
                data-value="estudo"
              >
                estudo
              </div>
              <div
                class="custom-option"
                data-testid="options"
                data-value="trabalho"
              >
                trabalho
              </div>
            </div>
            <select
              id="real-select"
            >
              <option
                value="finanças"
              >
                finanças
              </option>
              <option
                value="comida"
              >
                comida
              </option>
              <option
                value="estudo"
              >
                estudo
              </option>
              <option
                value="trabalho"
              >
                trabalho
              </option>
            </select>
          </div>
        </div>
        <div
          class="c10 primary undefined"
        >
          <div
            class="custom-select"
          >
            <label
              class="c11"
              for="priority"
            >
               
              Prioridade
            </label>
            <div
              class="custom-select-trigger primary undefined"
              data-testid="select-priority"
              id="priority"
            />
            <div
              class="custom-options primary undefined"
            >
              <div
                class="custom-option"
                data-testid="options"
                data-value="urgente"
              >
                urgente
              </div>
              <div
                class="custom-option"
                data-testid="options"
                data-value="alta"
              >
                alta
              </div>
              <div
                class="custom-option"
                data-testid="options"
                data-value="media"
              >
                media
              </div>
              <div
                class="custom-option"
                data-testid="options"
                data-value="baixa"
              >
                baixa
              </div>
            </div>
            <select
              id="real-select"
            >
              <option
                value="urgente"
              >
                urgente
              </option>
              <option
                value="alta"
              >
                alta
              </option>
              <option
                value="media"
              >
                media
              </option>
              <option
                value="baixa"
              >
                baixa
              </option>
            </select>
          </div>
        </div>
      </div>
      <div
        class="c12 primary undefined"
      >
        <label
          class="c13"
          for="description"
        >
          Descrição
        </label>
        <textarea
          class="c14 c15"
          id="description"
          name="description"
          title="message"
        />
      </div>
      <div
        class="c8"
      >
        <button
          class="c16 c17"
          type="submit"
        >
          Salvar
        </button>
        <button
          class="c16 c18"
          type="button"
        >
          Cancelar
        </button>
      </div>
    </form>
  </div>
   
</div>
`;
